packages/amplication-client/src/Entity/SchemaField.tsx

```diff
@@ -36,26 +36,33 @@ export const SchemaField = ({
   const fieldName = `properties.${propertyName}`;
   const label = propertySchema.title || capitalCase(propertyName);
 
-  const enumOptions = useMemo((): OptionItem[] | null => {
-    if (propertySchema.enum) {
-      return (propertySchema.enum as string[]).map((item) => {
-        const labelByItem: { label: string; value: string } =
-          ENTITY_FIELD_ENUM_MAPPER[fieldDataType][propertyName].find(
-            (option) => option.value === item
-          );
+  const enumOptions = useCallback(
+    function (enumOptions: string[]): OptionItem[] | null {
+      if (enumOptions) {
+        return enumOptions.map((item) => {
+          const labelByItem: { label: string; value: string } =
+            ENTITY_FIELD_ENUM_MAPPER[fieldDataType][propertyName].find(
+              (option) => option.value === item
+            );
 
-        return {
-          value: item,
-          label: labelByItem.label || item,
-        };
-      });
-    } else return null;
-  }, [fieldDataType, propertyName, propertySchema]);
+          return {
+            value: item,
+            label: labelByItem.label || item,
+          };
+        });
+      } else return null;
+    },
+    [fieldDataType, propertyName]
+  );
 
   if (propertySchema.enum) {
     if (propertySchema.enum.every((item) => typeof item === "string")) {
       return (
-        <SelectField label={label} name={fieldName} options={enumOptions} />
+        <SelectField
+          label={label}
+          name={fieldName}
+          options={enumOptions(propertySchema.enum as string[])}
```

Comment:
Is this casting necessary? `typeof` should already instruct TS compiler about the type.